airflow/www/fab_security/sqla/models.py

```diff
@@ -178,6 +180,13 @@ def is_active(self):
     def is_anonymous(self):
         return False
 
+    @property
+    def perms(self):
+        perms = set()
+        for role in self.roles:
+            perms.update((perm.action.name, perm.resource.name) for perm in role.permissions)
```

Comment:
I think you may need to do this as a single query for efficiency reasons?

Reply:
I added `lazy=joined` to several model fields, which causes prefetching. It now runs 11 queries instead of 49.

Reply:
@andrewgodwin that said, this PR does add some overall time to the tests. `pytest tests/views --with-db-init` runs in 1:53 on main and 2:00 on my branch. How significant is that?